demonstrations/tutorial_kernels_module.py

```diff
@@ -0,0 +1,623 @@
+thus determine its slope. The independent term :math:`b` specificies the
+position on the plane. In this form, linear classification can also be
+extended to higher dimensional vectors :math:`\boldsymbol{x}`, where a
+line does not divide the entire space into two regions anymore. Instead
+one needs a *hyperplane*. It is immediately clear that this method is
+not very powerful, as datasets that are not separable by a hyperplane
+can't be classified without error.
+
+We can actually sneak around this limitation by performing a neat trick:
+if we define some map :math:`\phi(\boldsymbol{x})` that *embeds* our
+datapoints into a larger *feature space* and then perform linear
+classification there, we could actually realise non-linear
+classification in our original space!
+
+.. figure:: ../demonstrations/kernels_module/embedding_nonlinear_classification.png
+   :align: center
+   :width: 65%
+
+If we go back to the expression for our prediction and include the
+embedding, we get
+
+.. math::
+   y(\boldsymbol{x}) = \operatorname{sgn}(\langle \boldsymbol{w}, \phi(\boldsymbol{x})\rangle + b).
+
+We will forgo one tiny step, but it can be shown that for the purpose
+of optimal classification, we can choose the vector defining the
+decision boundary as a linear combination of the embedded datapoints
+:math:`\boldsymbol{w} = \sum_i \alpha_i \phi(\boldsymbol{x}_i)`. Putting
+this into the formula yields
+
+.. math::
+   y(\boldsymbol{x}) = \operatorname{sgn}\left(\sum_i \alpha_i \langle \phi(\boldsymbol{x}_i), \phi(\boldsymbol{x})\rangle + b\right).
+
+This rewriting might not seem useful at first, but notice the above
+formula only contains inner products between vectors in the embedding
+space:
+
+.. math::
+   k(\boldsymbol{x}_i, \boldsymbol{x}_j) = \langle \phi(\boldsymbol{x}_i), \phi(\boldsymbol{x}_j)\rangle.
+
+We call this function the *kernel*. It provides the advantage that we can often
+find an explicit formula for the kernel :math:`k` that makes it
+superfluous to actually perform the (potentially expensive) embedding
+:math:`\phi`. Consider for example the following embedding and the
+associated kernel:
+
+.. math::
+   \phi((x_1, x_2)) &= (x_1^2, \sqrt{2} x_1 x_2, x_2^2) \\
+   k(\boldsymbol{x}, \boldsymbol{y}) &= x_1^2 y_1^2 + 2 x_1 x_2 y_1 y_2 + x_2^2 y_2^2 = \langle \boldsymbol{x}, \boldsymbol{y} \rangle^2.
+
+This means by just replacing the regular scalar product in our linear
+classification with the map :math:`k`, we can actually express much more
+intricate decision boundaries!
+
+This is very important, because in many interesting cases the embedding :math:`\phi`
+will be much costlier to compute than the kernel :math:`k`.
+
+In this demo, we will explore one particular kind of kernel
+that can be realized on near-term quantum computers, namely *Quantum
+Embedding Kernels (QEKs)*. These are kernels that arise from embedding
+data into the space of quantum states. We formalize this by considering
```

Comment:
Optional: I tend to prefer bolding new terms, since it stands out more than italics.

```suggestion
that can be realized on near-term quantum computers, namely **Quantum
Embedding Kernels (QEKs)**. These are kernels that arise from embedding
data into the space of quantum states. We formalize this by considering
```